File: solutions/julia/squeaky-clean/1/squeaky-clean.jl
function transform(ch)
    if ch == '-'
        "_"
    elseif ch == ' '
        ""
    elseif isuppercase(ch)
        "-" * lowercase(ch)
    elseif isdigit(ch)
        ""
    elseif ch >= 'α' && ch <= 'ω'
        "?"
    else
        string(ch)
    end
end

function clean(str)
    join([transform(ch) for ch in str])
end

File: solutions/fsharp/squeaky-clean/1/SqueakyClean.fs
module SqueakyClean

open System

let transform (c: char) : string =
    match c with
    | '-' -> "_"
    | ch when Char.IsWhiteSpace ch -> ""
    | ch when Char.IsUpper ch -> "-" + (Char.ToLower ch).ToString()
    | ch when Char.IsDigit ch -> ""
    | ch when 'α' <= ch && ch <= 'ω' -> "?"
    | _ -> c.ToString()

let clean (identifier: string) : string = String.collect transform identifier
